Sources/KituraNet/HTTP/HTTPHandler.swift

```diff
@@ -72,12 +73,15 @@ public class HTTPHandler: ChannelInboundHandler {
                 serverRequest.buffer!.byteBuffer.write(buffer: &buffer)
             }
         case .end(_):
-            serverResponse = HTTPServerResponse(ctx: ctx, handler: self)
+            serverResponse = HTTPServerResponse(channel: ctx.channel, handler: self)
             //Make sure we use the latest delegate registered with the server
-            if let delegate = server.delegate {
-                Monitor.delegate?.started(request: serverRequest, response: serverResponse)
-                delegate.handle(request: serverRequest, response: serverResponse)
-            } //TODO: failure path
+            DispatchQueue.global().async { [weak self] in
+                guard let weakSelf = self else { fatalError("No HTTPHandler to proceed with the response.") }
+                if let delegate = weakSelf.server.delegate {
+                    Monitor.delegate?.started(request: weakSelf.serverRequest, response: weakSelf.serverResponse)
+                    delegate.handle(request: weakSelf.serverRequest, response: weakSelf.serverResponse)
+                } //TODO: failure path
```

Comment:
Can we sort out this failure TODO now?

Reply:
Yeah, I think we can use the `HTTPDummyServerDelegate` here. We use that if no delegate is registered.